ddtrace/debugging/_encoding.py

```diff
@@ -289,36 +287,98 @@ def _get_fields(obj):
         return {s: _safe_getattr(obj, s) for s in get_slots(obj)}
 
 
-def _captured_value_v2(value, level=1):
-    # type: (Any, int) -> Dict[str, Any]
+def _captured_value_v2(value, level=0, maxlen=1024, maxsize=10, maxfields=10):
+    # type: (Any, int, int, int, int) -> Dict[str, Any]
     _type = type(value)
-    data = {
-        "type": _qualname(_type),
-        "value": _serialize(value, level),
-    }  # type: Dict[str, Any]
-    if _type not in BUILTIN_TYPES:
-        data["fields"] = (
-            {n: _captured_value_v2(v, level - 1) for n, v in _get_fields(value).items()} if level > 0 else None
+
+    if _type in BUILTIN_SIMPLE_TYPES:
+        value_repr = repr(value)
+        return (
+            {
+                "type": _qualname(_type),
+                "value": value_repr,
+            }
+            if len(value_repr) <= maxlen
```

Comment:
not sure this is what it was is expected for strings.
We need to clarify the RFC around this. there are two examples on with "notCapturedReason":"length" and another is truncated: true.

So, two comments:
1. in terms of usability - long strings should be truncated and not omitted ("notCapturedReason":"length").
2. if a value was truncated, I would also expect to have original length of the string - I recall we chatted about this but I forgot if we decided to not have it at the end.

/cc @jpbempel @OmerRaviv

Reply:
yes, long strings must be truncated and not omitted.
I have fixed one place in the RFC, but it shouldn't be any `"notCapturedReason": length` in the the snapshot, if truncated, it's `"truncated": true`

for the length/size, definitely I don't want to put it for all the strings. only if truncated

Reply:
Thanks for clarifying. Originally the string was also truncated, but I only noticed the example with the "notCapturedReason" in the RFC and missed the one with "truncated". I'l fix this!